src/Admin/BaseFieldDescription.php

```diff
@@ -418,40 +419,37 @@ public function getFieldValue($object, $fieldName)
             return $this->getFieldValue($child, substr($fieldName, $dotPos + 1));
         }
 
-        // prefer method name given in the code option
+        // NEXT_MAJOR: Remove this code.
         if ($this->getOption('code')) {
-            $getter = $this->getOption('code');
-
-            if (!method_exists($object, $getter)) {
-                @trigger_error(
-                    'Passing a non-existing method in the "code" option is deprecated'
-                    .' since sonata-project/admin-bundle 3.x and will throw an exception in 4.0.',
-                    \E_USER_DEPRECATED
-                );
+            @trigger_error(
+                'The "code" option is deprecated since sonata-project/admin-bundle 3.x.'
+                .' Use the "accessor" code instead',
+                \E_USER_DEPRECATED
+            );
 
-            // NEXT_MAJOR: Remove the deprecation and uncomment the next line.
-//                throw new \LogicException('The method "%s"() does not exist.', $getter);
-            } elseif (!\is_callable([$object, $getter])) {
+            $getter = $this->getOption('code');
+            if ($this->getOption('parameters')) {
                 @trigger_error(
-                    'Passing a non-callable method in the "code" option is deprecated'
-                    .' since sonata-project/admin-bundle 3.x and will throw an exception in 4.0.',
+                    'The option "parameters" is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0.',
                     \E_USER_DEPRECATED
                 );
 
-            // NEXT_MAJOR: Remove the deprecation and uncomment the next line.
-//                throw new \LogicException('The method "%s"() does not have public access.', $getter);
-            } else {
-                if ($this->getOption('parameters')) {
-                    @trigger_error(
-                        'The option "parameters" is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0.',
-                        \E_USER_DEPRECATED
-                    );
+                return $object->{$getter}(...$this->getOption('parameters'));
+            }
 
-                    return $object->{$getter}(...$this->getOption('parameters'));
-                }
+            return $object->{$getter}();
+        }
 
-                return $object->{$getter}();
-            }
+        // prefer the method or the property path given in the code option
+        $accessor = $this->getOption('accessor', $fieldName);
+        if (\is_callable($accessor)) {
+            return $accessor($object);
+        } elseif (!\is_string($accessor)) {
```

Comment:
so should this also allow `PropertyPathInterface`? Reading the exception message it seems so

Reply:
Indeed, fixed